controllers/update_status.go

```diff
@@ -624,9 +640,77 @@ func validateProcessGroup(ctx context.Context, r *FoundationDBClusterReconciler,
 	processGroupStatus.UpdateCondition(fdbv1beta2.PodFailing, failing, cluster.Status.ProcessGroups, processGroupStatus.ProcessGroupID)
 	processGroupStatus.UpdateCondition(fdbv1beta2.PodPending, false, cluster.Status.ProcessGroups, processGroupStatus.ProcessGroupID)
 
+	disableTaintFeature := false
+	for _, taintConfiguredKey := range cluster.Spec.AutomationOptions.Replacements.TaintReplacementOptions {
+		if *taintConfiguredKey.Key == "*" && *taintConfiguredKey.DurationInSeconds < 0 {
+			// Disable detecting or replacing taint key
+			disableTaintFeature = true
+			logger.Info("Disable taint feature", "Disabled", disableTaintFeature)
+			break
+		}
+	}
+
+	if !disableTaintFeature {
+		// Update taint status
+		updateTaintCondition(ctx, r, cluster, pod, processGroupStatus)
+	}
+
 	return nil
 }
 
+// updateTaintCondition checks pod's node taint label and update pod's taint-related condition accordingly
+func updateTaintCondition(ctx context.Context, r *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster,
+	pod *corev1.Pod, processGroupStatus *fdbv1beta2.ProcessGroupStatus) error {
+	logger := log.WithValues("namespace", cluster.Namespace, "cluster", cluster.Name, "reconciler", "updateStatus", "step", "updateTaintCondition")
+	node := &corev1.Node{}
```

Comment:
I'm not sure if we have the node object cached but the initial design proposed to fetch all nodes at once: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/docs/design/node_taint_eviction.md#proposed-design. Is there a benefit in doing it like this?

Reply:
it's consistent with the current style that update_status `validateProcessGroup` for each pod.

What's the benefit of fetching and caching all nodes? What if the info becomes stale (like pod moves to another node, making some cached node no longer belongs to the cluster)? What if the node info is updated between   when it's fetched and cached and when it's used?

Reply:
I share with the concern about making too many k8s api calls. I'm more concerned about the pressure put on k8s apiserver. 

Is there a way to use a single kubeapi call to get all nodes that belong to FDB cluster**S** ?